src/strategies/StrategyCompound.sol

```diff
@@ -0,0 +1,317 @@
+			uint256 diff = _supplied - _targetSupply;
+			if (redeemAmount > diff) redeemAmount = diff;
+
+			uint256 diffBorrowed = _borrowed - _targetBorrowed;
+			uint256 repayAmount = diffBorrowed < redeemAmount ? diffBorrowed : redeemAmount;
+
+			_redeem(redeemAmount);
+			_repay(repayAmount);
+			_supplied -= redeemAmount;
+			_borrowed -= repayAmount;
+
+			++iterations;
+			if (iterations > 25) revert SomethingWentWrong();
+		}
+	}
+
+	function _getSafeCollateralRatio(uint256 _marketCollateralRatio)
+		internal
+		view
+		returns (uint256 safeCollateralRatio)
+	{
+		return _marketCollateralRatio > buffer ? _marketCollateralRatio - buffer : 0;
+	}
+
+	function _getTargetSupply(uint256 _unleveraged, uint256 _safeCollateralRatio)
+		internal
+		pure
+		returns (uint256 targetSupply)
+	{
+		// if _safeCollateralRatio = 0, targetSupply = unleveraged (aka borrowed = 0)
+		if (_safeCollateralRatio == 0) return _unleveraged;
+		// 1 / (1 - c)
+		return _unleveraged.mulDivDown(1e18, 1e18 - _safeCollateralRatio);
+	}
+
+	/// @dev maximum amount that can be borrowed without going below _marketCol
+	function _getBorrowAmount(
+		uint256 _supplied,
+		uint256 _borrowed,
+		uint256 _marketCol
+	) internal pure returns (uint256) {
+		uint256 max = _supplied.mulDivDown(_marketCol, 1e18);
+		if (_borrowed >= max) return 0;
+		return max - _borrowed;
+	}
+
+	/// @dev maximum amount that can be redeemed without going below _marketCol
+	function _getRedeemAmount(
+		uint256 _supplied,
+		uint256 _borrowed,
+		uint256 _marketCol
+	) internal pure returns (uint256) {
+		uint256 min = _borrowed.mulDivUp(1e18, _marketCol);
+		if (_supplied <= min) return 0;
+		return _supplied - min;
+	}
+
+	/// @dev asset -> cToken
+	function _mint(uint256 _assetAmount) internal {
+		uint256 errorCode = cToken.mint(_assetAmount);
+		if (errorCode > 0) revert CompoundErrorCode(errorCode);
+	}
+
+	/// @dev cToken -> asset
+	function _redeem(uint256 _assetAmount) internal {
+		uint256 errorCode = cToken.redeemUnderlying(_assetAmount);
+		if (errorCode > 0) revert CompoundErrorCode(errorCode);
+	}
+
+	function _borrow(uint256 _assetAmount) internal {
+		uint256 errorCode = cToken.borrow(_assetAmount);
+		if (errorCode > 0) revert CompoundErrorCode(errorCode);
+	}
+
+	function _repay(uint256 _assetAmount) internal {
+		uint256 errorCode = cToken.repayBorrow(_assetAmount);
+		if (errorCode > 0) revert CompoundErrorCode(errorCode);
+	}
+
+	function _approve() internal {
+		// approve mint asset into cToken
+		asset.safeApprove(address(cToken), type(uint256).max);
+
+		_approveSwap();
+	}
+
+	function _unapprove() internal {
+		asset.safeApprove(address(cToken), 0);
+
+		_unapproveSwap();
+	}
+
+	function _unapproveSwap() internal {
+		COMP.safeApprove(address(swap), 0);
+	}
+
+	// approve swap COMP to asset
+	function _approveSwap() internal {
+		COMP.safeApprove(address(swap), type(uint256).max);
+	}
+}
```

Comment:
Add function to withdraw accidentally sent tokens?
https://github.com/stakewithus/unagii-vault-v2/blob/c76570746474e445960052eaf1cb9b1190ed18e2/contracts/strategies/StrategyCompLev.sol#L677-L682